proposals/22-04-localize-command.md

```diff
@@ -0,0 +1,331 @@
+  directory named `localized-{target}` in the same directory as `scope`
+
+and the flags are:
+* `-s`, `--scope`  
+  `scope`: optional root directory, files outside which kustomize is not allowed to copy and localize; if not specified,
+  takes on value of `target`
+* `-n`, `--no-verify`: do not verify that the outputs of `kustomize build` for `target` and `newDir` are the same after
+  localization
+
+The command creates a copy of the `target` kustomization and the local files that `target` references at `newDir`. We
+define the "files that `target` references" as:
+
+* kustomization files that `target` directly or transitively references
+* configuration files that referenced kustomization files reference
+* exec binaries of referenced KRM functions
+
+Here, configuration file means a non-kustomization yaml file. The command cannot run on `target`s that need
+the `--load-restrictor LoadRestrictionsNone` flag on `kustomize build`. The command only copies referenced files that
+reside inside `scope`.
+
+The command localizes the copy of `target` at `newDir` by downloading all remote files that `target`
+references. Users do not have executable permission for downloaded exec binaries that KRM functions reference and for
+each such exec binary, the command will print a warning message to that effect. 
+
+The command creates a new `localized-files` directory, next to the file that referenced the downloaded files, to hold
+said files. Inside `localized-files`, the downloads are located on path:
+
+<pre>
+<ins>domain</ins> / <ins>organization</ins> / <ins>repo</ins> / <ins>version</ins> / <ins>path/to/file/in/repo</ins>
+</pre>
+
+where `version` corresponds to the `ref` query string parameter in the url, though ideally `version` ia a stable tag as
+opposed to a branch.
+
+The command replaces remote references in `newDir` with the local paths of the downloaded files. To help ensure
+that `newDir` is a clean copy, the command additionally overwrites absolute path references into `target` to point
+to `newDir`.
+
+As a convenience to the user, in the absence of the `--no-verify` flag, the command automatically tries to
+run `kustomize build`, without any flags, on `target` and the localized `newDir` to compare their outputs. The command
+indicates success if the outputs match and throws an error with the diff summary otherwise. This check, however, is not
+useful for certain `target`s, including those that need flags to build. In these cases, the command prints next steps
+that users can follow to check the output themselves. For example, for `target`s that reference KRM functions with a
+remote exec binary, the command suggests the user:
+
+1. add executable permissions for the downloaded exec binaries in `newDir` **that the user trusts**
+2. run `kustomize build` with flags `--enable-alpha-plugins --enable-exec` and self-verify the outputs
```

Comment:
Sidenote: diffing two Kustomize directories would be a great thing for 'kustomize cfg' to be able to do if it doesn't already (I'm on mobile and don't recall)

Reply:
Natasha and I don't think `kustomize cfg` is currently capable of diffing. Natasha suggested I document your idea here: https://github.com/kubernetes-sigs/kustomize/issues/4090#issuecomment-1148043874